runtime/src/accounts_cache.rs

```diff
@@ -86,7 +86,7 @@ impl Deref for SlotCacheInner {
 
 #[derive(Debug, Clone)]
 pub struct CachedAccount {
-    pub account: Account,
+    pub account: AccountNoData,
```

Comment:
The root place this impacts many situations is here. The cache now holds AccountNoData, which shares its data behind an Arc. So, as an item is loaded from the cache, the data is never copied until it needs to be modified by someone. Cloning the account clones the non-data fields and addrefs the Arc to the data.

A subsequent change will attempt to move this AccountNoData struct to AccountNoDataInner and make AccountNoData have an Arc to an AccountNoDataInner. This would reduce copying of the rest of the Account contents (lamports, executable, owner, rent_epoch)